motionplan/armplanning/plan_manager.go

```diff
@@ -272,75 +275,32 @@ func (pm *planManager) planSingleAtomicWaypoint(
 }
 
 // planParallelRRTMotion will handle planning a single atomic waypoint using a parallel-enabled RRT solver.
-func (pm *planManager) planParallelRRTMotion(
+func (pm *planManager) planRRTMotion(
 	ctx context.Context,
 	wp atomicWaypoint,
 	endpointPreview chan *node,
 	solutionChan chan *rrtSolution,
 	maps *rrtMaps,
 ) {
-	var rrtBackground sync.WaitGroup
-	if maps == nil {
-		solutionChan <- &rrtSolution{err: errors.New("nil maps")}
-		return
-	}
-
 	// publish endpoint of plan if it is known
-	var nextSeed *node
 	if len(maps.goalMap) == 1 {
 		for key := range maps.goalMap {
-			nextSeed = key
-		}
-		if endpointPreview != nil {
-			endpointPreview <- nextSeed
-			endpointPreview = nil
+			endpointPreview <- key
+			// Dan: I think the caller, upon observing `endpointPreview`, will never check
+			// `solutionChan`. But preserving existing behavior for now.
 		}
 	}
 
 	// This ctx is used exclusively for the running of the new planner and timing it out.
-	plannerctx, cancel := context.WithTimeout(ctx, time.Duration(wp.mp.planOpts.Timeout*float64(time.Second)))
+	plannerctx, cancel := context.WithTimeout(ctx, time.Duration(wp.motionPlanner.planOpts.Timeout*float64(time.Second)))
 	defer cancel()
 
-	plannerChan := make(chan *rrtSolution, 1)
-
-	// start the planner
-	rrtBackground.Add(1)
-	utils.PanicCapturingGo(func() {
-		defer rrtBackground.Done()
-		wp.mp.rrtBackgroundRunner(plannerctx, &rrtParallelPlannerShared{maps, endpointPreview, plannerChan})
-	})
-
-	// Wait for results from the planner.
-	select {
-	case <-ctx.Done():
-		// Error will be caught by monitoring loop
-		rrtBackground.Wait()
-		solutionChan <- &rrtSolution{err: ctx.Err()}
-		return
-	default:
-	}
-
-	select {
-	case finalSteps := <-plannerChan:
-		// We didn't get a solution preview (possible error), so we get and process the full step set and error.
-		smoothChan := make(chan []*node, 1)
-		rrtBackground.Add(1)
-		utils.PanicCapturingGo(func() {
-			defer rrtBackground.Done()
-			smoothChan <- wp.mp.smoothPath(ctx, finalSteps.steps)
-		})
-
-		// Receive the newly smoothed path from our original solve, and score it
-		finalSteps.steps = <-smoothChan
-
-		solutionChan <- finalSteps
-		return
-
-	case <-ctx.Done():
-		rrtBackground.Wait()
-		solutionChan <- &rrtSolution{err: ctx.Err()}
-		return
-	}
+	// Dan: This is preserving original behavior. Where we'll return whatever steps we have in
+	// addition to an error.
+	finalSteps, err := wp.motionPlanner.rrtRunner(plannerctx, maps)
+	finalSteps.steps = wp.motionPlanner.smoothPath(ctx, finalSteps.steps)
```

Comment:
This was intended to keep the existing behavior. `smoothPath` (very not obviously) accepts nil steps and outputs nil. I didn't want to chase down right now whether calling code intentionally handled a case where non-zero steps and* an error were both returned.